jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -159,47 +203,153 @@ def _on_spectral_unit_changed(self, *args):
                                self.spectral_unit.selected,
                                sender=self))
 
-    @observe('flux_unit_selected')
-    def _on_flux_unit_changed(self, *args):
+    @observe('flux_or_sb_selected', 'flux_unit_selected', 'sb_unit_selected')
+    def _on_flux_unit_changed(self, msg):
+        flux_or_sb = None
+        current_y = self.spectrum_viewer.state.y_display_unit
 
-        yunit = _valid_glue_display_unit(self.flux_or_sb_unit.selected, self.spectrum_viewer, 'y')
+        data_collection_unit = ''
+        # need to determine the input spectrum units to disable the additional
+        # drop down and possiblity of translations in Specviz.
+        if (
+            len(self.app.data_collection) > 0 and
+            self.app.data_collection[0] and
+            self.app.config == 'specviz'
+        ):
+            if check_if_unit_is_per_solid_angle(self.app.data_collection[0].get_object()._unit):
+                data_collection_unit = 'Surface Brightness'
+                self.flux_or_sb_config_disabler = 'Flux'
+            else:
+                data_collection_unit = 'Flux'
+                self.flux_or_sb_config_disabler = 'Surface Brightness'
 
-        if self.spectrum_viewer.state.y_display_unit != yunit:
+        # skip if arg is not a dictionary
+        if not isinstance(msg, dict):
```

Comment:
It had to do with this comment earlier, it was when I was originally using `*args` instead of `msg`. If this isn't necessary (and the below comment for the other similar check) I will remove them. 

https://github.com/spacetelescope/jdaviz/pull/2940#discussion_r1682702371

Reply:
I still don't think the checks are needed but I defer to @kecnry .

Reply:
I think its probably safe to remove now that its clear that the input is from the traitlet.  If the method can run (for debugging, etc) without the information from the traitlet, then we often set the default to `msg={}` in the call signature, but it looks like in this case the message is required and is required to have `'name'` defined, so that probably doesn't make sense here.